### PR TITLE
Update README.md documentation

### PR DESCRIPTION
**Pull request by miguelsorianod**
I added in the README.md how to deploy an empty 3scale-operator, and refer to the user-guide on more details on what it does and how to deploy some example custom resources for it.

### CODE REVIEW

**Comment by mikz** [comment]
Why into README.md ? Bloated readmes are quite off putting. Each part of the readme can be independent document and the README acts as just as a table of contents.

**Comment by miguelsorianod** [comment]
I agree. For the moment I rolled back the user-guide content, kept the contributing section and added a reference to the user-guide from the quickstart.

**Comment by miguelsorianod** [comment]
How about now?

File: README.md [hunk 1]
```diff
@@ -1,25 +1,77 @@
...
+Download the 3scale-operator project:
 
 ```sh
-$ mkdir -p $GOPATH/src/github.com/3scale
-$ cd $GOPATH/src/github.com/3scale
-$ git clone https://github.com/3scale/3scale-operator
-$ cd 3scale-operator
-$ git checkout master
-$ make vendor
+mkdir -p $GOPATH/src/github.com/3scale
```
Comment (mikz) [inline README.md]: Can't you just `go get -u && cd`?

File: README.md [hunk 2]
```diff
@@ -1,25 +1,77 @@
...
 
+To see an example deploy of the 3scale-operator and its CRDs refer
+to the following [User guide](doc/user-guide.md)
+
+## Development
+
+Download the 3scale-operator project and install the needed dependencies:
+
+```sh
+mkdir -p $GOPATH/src/github.com/3scale
```
Comment (mikz) [inline README.md]: This is the same as above. Should not be needed to repeat.

Comment (miguelsorianod) [reply README.md]: There was one additional step. I modified the content to refer to the other section from here and just add the additional step

File: README.md (hunk 1 above)
Comment (miguelsorianod) [reply README.md]: It seems that the structure of the project is not completely go-gettable (the project structure is generated by the operator-sdk new command):

```
msoriano@localhost:~$ go get -u github.com/3scale/3scale-operator
package github.com/3scale/3scale-operator: no Go files in /home/msoriano/exampleworkspace/src/github.com/3scale/3scale-operator
```

I've seen other projects doing it the way I wrote too (operator-sdk for example), maybe for the same reason.

**Comment by miguelsorianod** [comment]
I've updated the PR again.
I've now added in the README.md how to deploy an empty 3scale-operator, and refer to the user-guide on more details on what it does and how to deploy some example custom resources for it.

The idea is to add the remaining Capabilities part in the user-guide @jmprusi  in another PR, and also the corresponding Capabilities-related spec file/s and reference them from the README.md.

**Pull request merged**